modules/libcom/src/osi/os/default/osdEnv.c

```diff
@@ -20,40 +20,27 @@
 #include <errno.h>
 
 #include "epicsStdio.h"
+#include "epicsVersion.h"
 #include "errlog.h"
-#include "cantProceed.h"
 #include "envDefs.h"
 #include "osiUnistd.h"
 #include "epicsFindSymbol.h"
 #include "iocsh.h"
 
+#ifdef __rtems__
+#  define RTEMS_VERSION_INT VERSION_INT(__RTEMS_MAJOR__, __RTEMS_MINOR__, 0, 0)
+#endif
+
```

Comment:
The whole point of our having the `osi/os/$(OS_CLASS)/` source structure was to avoid putting OS-specific conditionals into the source files since that makes them harder to read. The differences here are pretty minor so I understand the desire to not duplicate most of this code in an `osi/os/RTEMS/osdEnv.c` file, but they could be encapsulated even better like this:

```C
#ifdef __rtems__
#  define RTEMS_VERSION_INT VERSION_INT(__RTEMS_MAJOR__, __RTEMS_MINOR__, 0, 0)
#endif

#if defined(__rtems__) && RTEMS_VERSION_INT<VERSION_INT(4,10,0,0)
   /* newlib w/ RTEMS <=4.9 returns void */
#  define unsetenv_M(name) ({unsetenv(name); 0})
#else
#  define unsetenv_M(name) unsetenv(name)
#endif
```
Then remove the RTEMS-4.9 conditional below and call `unsetenv_M()` instead of `unsetenv()`.

Note that there is also a comment about `putenv()` immediately above the `epicsEnvUnset()` routine below which needs to be removed.

Reply:
Right, I hardly ever use the `({...})` construct, but I guess it should be compatible with old enough gcc and clang.  Would it work, and does it need to work, with suncc?

Digression...

I was thinking about putting the (conditional) definition of `RTEMS_VERSION_INT` in `epicsVersion.h`.  Maybe also `GCC_VERSION_INT`.  We use these somewhat widely, and the definitions should always be the same.

Reply:
The `({ ... })` construct is only needed when building for RTEMS <= 4.9, the other branch of my suggestion doesn't need/use it. Are you asking whether RTEMS 4.9 was built with suncc? 😉 

Exporting `RTEMS_VERSION_INT` might have a deleterious effect on external modules which also define it and would not expect it to be already defined. I'm not objecting to the idea, but it might have some repercussions downstream. There is only one place each where we currently define and use that and `VXWORKS_VERSION_INT` although I haven't checked the RTEMS-5 branch.

Reply:
> ... Are you asking whether RTEMS 4.9 was built with suncc? wink

Brain fart...

> Exporting `RTEMS_VERSION_INT` might have a deleterious effect ...

True, although at worst this would force adding another `#ifdef` (which I would include in `epicsVersion.h` as well).  I see both the benefits and costs as minor.  So I may add these definitions the next time it would be convenient, or not.